### PR TITLE
Check for min vCenter version

### PR DESCRIPTION
**Pull request by dvonthenen**
**What this PR does / why we need it**:
This checks the minimum vCenter API version (which is 6.5) on the CSI FCD controller. If unsupported, the controller will print an error message and will fail to start the controller service.

I had hopes that we could check clusters within DCs, but quickly realized that there could be clusters that are being used to manage non-Kubernetes earmarked clusters that can be of an older ESX version. Only check that can be done without going through a bunch of hoops is checking the vCenter API version. Docs already outline the support matrix for the CSI driver.

**Which issue this PR fixes**:
https://github.com/kubernetes/cloud-provider-vsphere/issues/139

**Special notes for your reviewer**:
Added go tests to check the boundary cases.

Tested CCM and CSI on:
- k8s 1.13.1 cluster with 10 worker nodes in 3 zones
- vSphere 6.7u1

**Release note**:
NA

### CODE REVIEW

**Comment by dvonthenen** [comment]
CC: @frapposelli @dougm @codenrhoden

File: pkg/csi/service/fcd/vsphere_helper.go [hunk 1]
```diff
@@ -31,8 +33,35 @@ import (
 const (
 	NUM_OF_CONNECTION_ATTEMPTS     int = 3
 	RETRY_ATTEMPT_DELAY_IN_SECONDS int = 1
+
+	MINIMUM_SUPPORTED_VCENTER_MAJOR int = 6
+	MINIMUM_SUPPORTED_VCENTER_MINOR int = 5
 )
 
+func APICheck(version string) error {
```
Comment (frapposelli) [inline pkg/csi/service/fcd/vsphere_helper.go]: I see this is exported, is it going to be used in other packages?

File: pkg/common/connectionmanager/connectionmanager.go [hunk 2]
```diff
@@ -188,3 +188,17 @@ func (cm *ConnectionManager) VerifyWithContext(ctx context.Context) error {
 	}
 	return nil
 }
+
+func (cm *ConnectionManager) APIVersion(vcenter string) (string, error) {
```
Comment (frapposelli) [inline pkg/common/connectionmanager/connectionmanager.go]: can you add a comment here? to keep `go vet` happy 😄

File: pkg/common/connectionmanager/connectionmanager.go [hunk 3]
```diff
@@ -188,3 +188,17 @@ func (cm *ConnectionManager) VerifyWithContext(ctx context.Context) error {
...
+	err := cm.Connect(context.Background(), vcenter)
```
Comment (frapposelli) [inline pkg/common/connectionmanager/connectionmanager.go]: nit: can you make it terse with
```go
if err := cm.Connect(context.Background(), vcenter); err != nil {
```

File: pkg/csi/service/fcd/controller.go [hunk 4]
```diff
@@ -71,6 +72,21 @@ func (c *controller) Init(config *vcfg.Config) error {
...
 
+	//VC check... FCD is only supported in 6.5+
+	for vc := range connMgr.VsphereInstanceMap {
+		api, err := connMgr.APIVersion(vc)
+		if err != nil {
+			klog.Errorf("APIVersion failed err=%v", err)
+			return err
+		}
+
+		err = APICheck(api)
```
Comment (frapposelli) [inline pkg/csi/service/fcd/controller.go]: nit: terse
```go
if err := APICheck(api); err != nil {
```

File: pkg/csi/service/fcd/vsphere_helper.go (hunk 1 above)
Comment (dvonthenen) [reply pkg/csi/service/fcd/vsphere_helper.go]: Nope... will rename the function. Was complaining that "API" was in lowercase.

Comment (frapposelli) [reply pkg/csi/service/fcd/vsphere_helper.go]: maybe `checkAPI`?

File: pkg/common/connectionmanager/connectionmanager.go (hunk 2 above)
Comment (dvonthenen) [reply pkg/common/connectionmanager/connectionmanager.go]: Will do

File: pkg/common/connectionmanager/connectionmanager.go (hunk 3 above)
Comment (dvonthenen) [reply pkg/common/connectionmanager/connectionmanager.go]: Yup

File: pkg/csi/service/fcd/controller.go (hunk 4 above)
Comment (dvonthenen) [reply pkg/csi/service/fcd/controller.go]: same

**Comment by dvonthenen** [comment]
Updated based on feedback. Thanks @frapposelli

**Comment by frapposelli** [comment]
cool stuff, thanks @dvonthenen !

/approve
/lgtm
